src/iuse_actor.cpp

```diff
@@ -1799,6 +1799,27 @@ long holster_actor::use( player *p, item *it, bool, const tripoint & ) const
 
     if( pos >= 0 ) {
         p->wield_contents( it, pos, draw_speed );
+
+        // handle iaijutsu attacks when drawing swords and other such blades
+        if( p->weapon.has_flag( "IAIJUTSU" ) &&
+            p->skillLevel( skill_id("cutting") ) >= 7 &&
+            one_in(12 - p->skillLevel( skill_id("cutting") ) ) ) {
+
+            for( const auto &pos : closest_tripoints_first( 1, p->pos() ) ) {
+                if( g->mon_at( pos, true ) != -1 ) {
+                    // at least one adjacent monster
+                    tripoint sel;
+                    if (choose_adjacent( _( "Slash where?" ), sel ) && g->mon_at( sel, true ) != -1 ) {
```

Comment:
It might be better to use `game::critter_at` instead of `mon_at`, at least in new code. The former returns references to any kind of creatures (including the player, so this may have to be checked). This would allow to use this attack against NPCs as well.